src/TritonRoute/src/ta/FlexTA_assign.cpp

```diff
@@ -94,8 +94,8 @@ void FlexTAWorker::modMinSpacingCostPlanar(const frBox &box, frLayerNum lNum, ta
     xform.set(frPoint(boxLeft, trackLoc));
     box2.transform(xform);
     box2boxDistSquare(box1, box2, dx, dy);
-    frCoord maxX = (frCoord)(sqrt(1.0 * bloatDistSquare - 1.0 * dy * dy));
-    if (maxX * maxX + dy * dy == bloatDistSquare) {
+    frCoord maxX = (frCoord)(sqrt(1.0 * bloatDistSquare - 1.0 * (frSquaredDistance)dy * dy));
```

Comment:
This one is unnecessary but not harmful as dy will be converted to a double by 1.0*dy with or without the cast